Modules/core/Controller/CoretilesController.php

```diff
@@ -99,4 +109,91 @@ public function docAction(){
         ));
     }
 
+
+    /**
+     * Distinctly list spaces:
+     * - of which user is member
+     * - in which user has a pending request to join
+     * - of which user is admin
+     * 
+     * @return array of arrays: [userSpaceIds, userPendingSpaceIds, SpacesUserIsAdminOf]
+     */
+    public function getUserSpaces() {
+        $modelSpacePending = new CorePendingAccount();
+        $data = $modelSpacePending->getSpaceIdsForPending($_SESSION["id_user"]);
+        $userPendingSpaceIds = array();
+
+        if ($data && count($data) > 0) {
+            foreach ($data as $space) {
+                array_push($userPendingSpaceIds, $space["id_space"]);
+            }
+        }
+
+        $modelSpaceUser = new CoreSpaceUser();
+        $data = $modelSpaceUser->getUserSpaceInfo($_SESSION["id_user"]);
```

Comment:
invalid getuserspaceinfo usage ? this method returns one (and only one) user/space relation
or there is a bug in methods:

        $sql = "SELECT * FROM core_j_spaces_user WHERE id_user=?";
        return $this->runRequest($sql, array($id_user))->fetch();

should it be a fetchAll (get all spaces user is a member of)